File: NeetCode150/Two Pointers/11. Container With Most Water/sol.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0, right = height.size()-1;
        int maxVolume = 0;
        int currentMax = 0;

        while (left < right) {
            //看前後比對的哪一個大
            if (height[left] >= height[right]) {
                currentMax = height[right] * (right-left);
            }
            else {
                currentMax = height[left] * (right-left);
            }
            //若體積>目前最大就更新最大值
            if (currentMax > maxVolume) {
                maxVolume = currentMax;
            }
            //左指標往右移 右指標往左移(第一次寫的時候沒注意到)
            if (height[left] <= height[right]) {
                left++; 
            } else {
                right--;
            }
        }
        return maxVolume;
    }
};

File: NeetCode150/Two Pointers/11. Container With Most Water/sol.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        l, r = 0, len(heights) - 1

        maxSum = 0
        curSum = 0
        while l < r:
            if heights[l] <= heights[r]:
                curSum = heights[l] * (r - l)
                l += 1
            else:
                curSum = heights[r] * (r - l)
                r -= 1
            maxSum = max(curSum, maxSum)
        return maxSum